### ISSUE TITLE
ValueFormatter get should at least pass datasetIndex

### ISSUE THREAD

**Issue by filippodelfra** [PhilJay/MPAndroidChart#4490]
Hi,
I'm asking if it's possible to add the index of the data set entry to the getFormattedValue() method. 
My implementation was using the entry because we are not showing the value that is used to generate the chart.
So, since in version 3.1.0 the method getFormattedValue(float value, Entry entry, int dataSetIndex, ViewPortHandler viewPortHandler) has been deprecated, I feel that we should at least have the index. 

Thanks!

**Comment by ClaireGizard** [comment]
Hi !  
I have the same problem, did you find a solution to get the index ?
Thank's,
Claire

**Comment by filippodelfra** [comment]
Hi,
No I'm still on 3.0 branch for now.

**Comment by waqdon** [comment]
Hi !
I have the same problem, too, did you find a solution to get the index?
Thank you

**Comment by yaroslav-shlapak** [comment]
Same here, the deprecated method is not called by the library.
Are there any other options rather than downgrading to 3.0.0?

**Comment by GokhanArik** [comment]
I agree with you all. Have you tried overriding `getPointLabel() : Entry` and getting index from entries?

**Comment by GokhanArik** [comment]
It would be really helpful to have this. 
I know this is not ideal, but I can recommend a temporary fix.
Assign a key to your data set and set it to `data` field on `Entry` object. Or you can assign position to that field.
In ValueFormatter override `getPointLabel(entry: Entry)`, and retrieve `data` field.
